tests/extract/test_extract_pipe.py

```diff
@@ -715,13 +717,14 @@ async def long_gen():
                 yield i
             close_pipe_yielding = False
         # we have a different exception here
-        except asyncio.CancelledError:
+        except GeneratorExit:
```

Comment:
I don't think this works. the test fails now. The error on cancelling async gens is a cancellederror, not a generatorexit.

Reply:
huh this was working for me. I was not cancelling async gen directly but the wrapped gen and propagating the exception. I'll check it out